pilot/pkg/serviceregistry/kube/controller/endpointcontroller.go

```diff
@@ -116,3 +118,31 @@ func updateEDS(c *Controller, epc kubeEndpointsController, ep interface{}, event
 		}
 	}
 }
+
+func getPod(c *Controller, ip string, ep *metav1.ObjectMeta, targetRef *v1.ObjectReference, host host.Name) *v1.Pod {
+	pod := c.pods.getPodByIP(ip)
+	if pod != nil {
+		return pod
+	}
+	// This means, the endpoint event has arrived before pod event.
+	// This might happen because PodCache is eventually consistent.
+	if targetRef != nil && targetRef.Kind == "Pod" {
+		key := kube.KeyFunc(targetRef.Name, targetRef.Namespace)
```

Comment:
This looks right but I don't understand how the test passes with and without this change?

Reply:
Yes. I will check the problem with the test